lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb

```diff
@@ -191,7 +193,29 @@ def self.run_release_upload(params)
             release = Helper::AppcenterHelper.update_release(api_token, owner_name, app_name, release_id, release_notes)
             Helper::AppcenterHelper.update_release_metadata(api_token, owner_name, app_name, release_id, dsa_signature)
 
-            destinations_array = destinations.split(',')
+            destinations_array = []
+            if destinations == '*'
+              UI.message("Looking up all distribution groups for #{owner_name}/#{app_name}")
+              distribution_groups = Helper::AppcenterHelper.fetch_distribution_groups(
+                api_token: api_token,
+                owner_name: owner_name,
+                app_name: app_name
+              )
+
+              destination_excludes_array = destination_excludes.split(',').map(&:strip).map(&:downcase)
+              UI.abort_with_message!("Failed to list distribution groups for #{owner_name}/#{app_name}") unless distribution_groups
+              distribution_groups.each do |group|
+                destinations_array << group['name'] unless destination_excludes_array.include? group['name'].downcase 
+              end
+
+              distribution_group_names = distribution_groups.map {|h| h['name'].downcase }
+              destination_excludes_array.each do |group|
+                UI.important("Could not find the excluded distribution group: #{group}") unless distribution_group_names.include? group.downcase 
+              end
+            else
+              destinations_array = destinations.split(',')
```

Comment:
It might be worth it adding a `map(&:strip)` here as well while we're at it.

Reply:
map(&:strip) added as well as unit tests